### PR TITLE
fix: Color.ToString now returns the expected format

### PR DESCRIPTION
**Pull request by VitezslavImrysek**
GitHub Issue (If applicable): #5189

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
Color.ToString() returns in format '[#AABBCCDD]' which is incorrect.

## What is the new behavior?
Color.ToString() now returns in format '#AABBCCDD'.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]
🏎️

**Comment by MartinZikmund** [comment]
The change of `IFormattable` to explicitly implemented is potentially breaking, but it is very unlikely someone was using the method anyway and it matches UWP.

**Review by jeromelaban — APPROVED** [review]
Thanks for your contribution!

**Review by ajpinedam — APPROVED** [review]

**Comment by robloo** [comment]
Shouldn't this somehow be fixed to the InvariantCulture? Allowing a format provider might theoretically cause invalid formats. Although, I'm not sure of any culture where it would cause an issue.

File: src/Uno.UWP/UI/Color.cs [hunk 1]
```diff
@@ -34,14 +35,26 @@ private Color(byte a, byte r, byte g, byte b)
...
+		string IFormattable.ToString(string format, IFormatProvider formatProvider) => ToString(format, formatProvider);
+
+		private string ToString(string format, IFormatProvider formatProvider)
+		{
+			var sb = new StringBuilder();
+
+			sb.AppendFormat(formatProvider, "#{0:X2}", A);
+			sb.AppendFormat(formatProvider, "{0:X2}", R);
+			sb.AppendFormat(formatProvider, "{0:X2}", G);
+			sb.AppendFormat(formatProvider, "{0:X2}", B);
```
Comment (carldebilly) [inline src/Uno.UWP/UI/Color.cs:55]: Is there any avantage doing that over a simple `string.Format(formarProvider, "#{0:X2}{1:X2}{2:X2}{3:X2}", A, R, G, B)`?

**Review by carldebilly — APPROVED** [review]

**Comment by jeromelaban** [comment]
Some tests were relying on the string to validate the content. Those will need to be adjusted as well. You can find examples here: 

https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=30721&view=ms.vss-test-web.build-test-results-tab&runId=1502890&resultId=100067&paneView=debug

File: src/Uno.UWP/UI/Color.cs (hunk 1 above)
Comment (VitezslavImrysek) [reply src/Uno.UWP/UI/Color.cs]: I am not aware of any. I changed it to your suggestion.

**Pull request merged**